crates/feldera-storage/src/lib.rs

```diff
@@ -8,17 +8,26 @@ pub mod file;
 mod init {
     use ctor::ctor;
     use fdlimit::raise_fd_limit;
+    use fdlimit::Outcome::LimitRaised;
+    use log::warn;
 
     /// Raise the fd limit to run the storage library to avoid surprises.
     ///
     /// Note that this is a no-op on Windows platforms, hence it's not behind an
-    /// architecture cfg.
-    ///
-    /// TODO: We should raise a warning/abort if the fd limit is (still) too low
-    /// e.g., due to hard-limit being set too low and display a message on
-    /// how to fix it.
+    /// architectural cfg.
     #[ctor]
     fn init() {
-        let _r = raise_fd_limit();
+        match raise_fd_limit() {
+            Ok(LimitRaised { from, to }) => {
+                const WARN_THRESHOLD: u64 = 2 << 19;
```

Comment:
`2 <<  19` looks weird. Probably you were thinking `2^19`, which is written as `1 << 19`.